vowpalwabbit/cb_adf.cc

```diff
@@ -269,14 +271,14 @@ void gen_cs_example_MTR(cb_adf& data, v_array<example*>& ec_seq, v_array<example
 
 template<bool predict>
 void learn_MTR(cb_adf& mydata, base_learner& base, v_array<example*>& examples)
-{ uint32_t action = 0;
+{ //uint32_t action = 0;
   if (predict) //first get the prediction to return
   { gen_cs_example_ips(examples, mydata.cs_labels);
     call_predict_or_learn<false>(mydata, base, examples, mydata.cb_labels, mydata.cs_labels);
-    if (!mydata.rank_all) //preserve prediction
-      action = examples[0]->pred.multiclass;
-    else
-      std::swap(examples[0]->pred.a_s, mydata.a_s);
+    //    if (!mydata.rank_all) //preserve prediction
+    //      action = examples[0]->pred.multiclass;
+    //    else
```

Comment:
These comments were mostly left for John to see since they involved
modifying the return type from cb_adf when --rank_all is not specified.
They can be removed.

Alekh

On Tue, Jun 7, 2016 at 10:26 AM, Luong Hoang <notifications@github.com>
wrote:

> In vowpalwabbit/cb_adf.cc
> <https://github.com/JohnLangford/vowpal_wabbit/pull/1006#discussion_r66080711>
> :
>
> >    if (predict) //first get the prediction to return
> >    { gen_cs_example_ips(examples, mydata.cs_labels);
> >      call_predict_or_learn<false>(mydata, base, examples, mydata.cb_labels, mydata.cs_labels);
> > -    if (!mydata.rank_all) //preserve prediction
> > -      action = examples[0]->pred.multiclass;
> > -    else
> > -      std::swap(examples[0]->pred.a_s, mydata.a_s);
> > +    //    if (!mydata.rank_all) //preserve prediction
> > +    //      action = examples[0]->pred.multiclass;
> > +    //    else
>
> same as above
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/JohnLangford/vowpal_wabbit/pull/1006/files/73eb39a39cbabe685f34de771a5b2469bcda7f3c#r66080711>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAyIGS0qZtfuOo0rKa0smxdh6VeNVhrLks5qJX-kgaJpZM4Iu-Dw>
> .
>